dist/tools/coverage/coverage.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+"""This script will parse the serial console log file and create the required
+gcda files.
+"""
+
+import argparse
+import os
+import re
+
+
+def retrieve_data(input_file):
+    extracted_coverage_info = {}
+    capture_data = False
+    reached_end = False
+    with open(input_file, 'r') as fp:
+        for line in fp.readlines():
```

Comment:
```suggestion
    with open(input_file, 'r') as fp:
        for line_bytes in fp.readlines():
            line = line_bytes.decode(errors="ignore")
```

should get rid of the UTF-8 decoding errors you reported. If the UTF-8 characters are not in fact invalid, one could always use another [error handler](https://docs.python.org/3/library/codecs.html#error-handlers).